userland/libtock/crt0.c

```diff
@@ -6,23 +6,78 @@ extern int main(void);
 #pragma GCC diagnostic ignored "-Wmissing-declarations"
 #pragma GCC diagnostic ignored "-Wmissing-prototypes"
 
+struct hdr {
```

Comment:
A comment here about this header structure being populated at the beginning of the binary by the linker would be useful.